devicetypes/smartthings/zigbee-multi-button.src/zigbee-multi-button.groovy

```diff
@@ -223,3 +254,77 @@ private getModelBindings(model) {
 	}
 	bindings
 }
+
+private Map parseAduroSmartButtonMessage(Map descMap){
+    def buttonState = ""
+    def buttonNumber = 0
+    if (descMap.clusterInt == 0x0006) {
+        log.debug "descMap 0x0006 is $descMap"
+        buttonState = "pushed"
+        if (descMap.command == "01") {
+            buttonNumber = 1
+        }
+        else if (descMap.command == "00") {
+            buttonNumber = 4
+        }
+        if (buttonNumber !=0) {
+            def childevent = createEvent(name: "button", value: "pushed",data: [buttonNumber: 1], isStateChange: true)
+            sendEventToChild(buttonNumber, childevent)
+            def descriptionText = "$device.displayName button $buttonNumber was $buttonState"
+            return createEvent(name: "button", value: buttonState, data: [buttonNumber: buttonNumber], descriptionText: descriptionText, isStateChange: true)
+        }
+        else {
+            return [:]
+        }
+    }
+    else if (descMap.clusterInt == 0x0008) {
+        if (descMap.command == "02") {
+            buttonState = "pushed"
+            def data = descMap.data
+            log.debug "descMap 0x0008 is $descMap"
+            def d0 = data[0]
+            //log.debug "d0 is $d0"
+            if (d0 == "00") {
+                buttonNumber = 2
+            }
+            if (d0 == "01") {
+                buttonNumber = 3
+            }
+            //state.buttonNumber = buttonNumber
+            //getButtonResult("press", buttonNumber)
+            state.buttonNumber = buttonNumber
+            state.pressTime = now()
+            def childevent = createEvent(name: "button", value: "pushed",data: [buttonNumber: 1], isStateChange: true)
+            sendEventToChild(buttonNumber, childevent)
+            return createEvent(name: "button", value: buttonState, data: [buttonNumber: buttonNumber], descriptionText: descriptionText, isStateChange: true)
```

Comment:
you duplicate these exact 3 lines of code in 3 separate if statements instead of just putting it after them